catkit2/services/thorlabs_cubes/thorlabs_cubes.py

```diff
@@ -0,0 +1,108 @@
+from catkit2.testbed.service import Service
+
+import thorlabs_apt as apt
+import numpy as np
+
+
+class ThorlabsCubes(Service):
+
+    def __init__(self):
+        super().__init__('thorlabs_cubes')
+
+        self.cube_serial_number = self.config['serial_number']
+
+        self.command = self.make_data_stream('command', 'float64', [1], 20)
+        self.current_position = self.make_data_stream('current_position', 'float64', [1], 20)
+
+        # Initialization of the parameters
+        self.motor = None
+        self.cube_model = None
+        self.motor_unit = None
+        self.motor_max = None
+        self.motor_min = None
+
+    def open(self):
+        """
+            Open the motor and cube.
+
+            This will also check if the configured characteristics are matching the one the cube have.
+        """
+        self.motor = apt.Motor(self.cube_serial_number)
+
+        # Retrieve min, max, unit and model of the motor and cube
+        min = self.motor.get_stage_axis_info()[0]
+        max = self.motor.get_stage_axis_info()[1]
+        unit = ''
+        model = self.motor.hardware_info[0][2:-1]
+        if self.motor.get_stage_axis_info()[2] == 1:
+            unit = 'mm'
+        if self.motor.get_stage_axis_info()[2] == 2:
+            unit = 'deg'
+
+        # Config min, max, unit and model of the motor and cube
+        self.min_pos = self.config['motor_min_pos']
+        self.max_pos = self.config['motor_max_pos']
+        self.unit = self.config['motor_unit']
+        self.cube_model = self.config['cube_model']
+
+        # Compare
+        try:
+            self.min_pos == min
+            self.max_pos == max
+            self.unit == unit
+            self.cube_model == model
+        except Exception:
+            raise

```

Comment:
Just because the equality checks return `False` doesn't mean that an exception will be raised, which is what you actually want here. Replace the `try` with an `if` and the `except` with an
```python
else:
   ValueError('Device parameters don't match configuration parameters.)
```
and you will get what you want.

Also, this is the type of thing you should test on hardware while developing. But some garbage into the respective values in the configfile and try to start the service. If you get the error you planted there, everything is fine.